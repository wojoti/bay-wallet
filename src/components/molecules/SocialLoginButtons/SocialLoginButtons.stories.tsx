import {ComponentMeta, ComponentStory} from '@storybook/react-native';
import {SocialLoginButtons} from 'components/molecules';

export default {
  title: 'Atoms/SocialLoginButtons',
  component: SocialLoginButtons,
  argTypes: {
    onIconPress: {action: 'onIconPress'},
  },
} as ComponentMeta<typeof SocialLoginButtons>;
const Template: ComponentStory<typeof SocialLoginButtons> = args => (
  <SocialLoginButtons {...args} />
);

export const Example = Template.bind({});
Example.args = {
  testId: '',
};
