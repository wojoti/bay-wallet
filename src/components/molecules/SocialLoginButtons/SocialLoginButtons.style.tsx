import {StyleSheet} from 'react-native';

const style = StyleSheet.create({
  width: {
    width: '47%',
  },
});
export default style;
