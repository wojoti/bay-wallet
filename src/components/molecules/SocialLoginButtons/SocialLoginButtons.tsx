import {ComposableButton, Container, Img, ImgType} from 'components/atoms';
import {PropsWithChildren} from 'react';
import style from './SocialLoginButtons.style';

export type SocialLoginButtonsProps = PropsWithChildren<{
  onIconPress: (origin: string) => void;
  testId?: string;
}>;

const SocialLoginButtons = ({onIconPress, testId}: SocialLoginButtonsProps) => {
  const onPressF = () => {
    onIconPress('facebook');
  };
  const onPressG = () => {
    onIconPress('google');
  };
  return (
    <Container
      direction="row"
      justify="space-between"
      align="flex-start"
      testId={testId}>
      <Container customStyle={style.width}>
        <ComposableButton
          customStyle={style.socialBtnStyle}
          onPress={onPressG}
          testId="socialloginbuttons-button-google">
          <Img
            src={ImgType.google}
            height="25"
            alt="google login"
            testId="img-google"
          />
        </ComposableButton>
      </Container>
      <Container customStyle={style.width}>
        <ComposableButton
          customStyle={style.socialBtnStyle}
          onPress={onPressF}
          testId="socialloginbuttons-button-facebook">
          <Img
            src={ImgType.facebook}
            height="25"
            alt="facebook login"
            testId="img-facebook"
          />
        </ComposableButton>
      </Container>
    </Container>
  );
};

export default SocialLoginButtons;
