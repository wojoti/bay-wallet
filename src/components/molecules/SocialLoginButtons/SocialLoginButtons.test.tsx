import '@testing-library/jest-native/extend-expect';
import {fireEvent, render, screen, within} from '@testing-library/react-native';
import TestRenderer from 'react-test-renderer';
import SocialLoginButtons, {
  SocialLoginButtonsProps,
} from './SocialLoginButtons';

const mockOnIconPress = jest.fn<
  SocialLoginButtonsProps['onIconPress'],
  [origin: string]
>();
const props: SocialLoginButtonsProps = {
  testId: 'test-socialloginbuttons-id',
  onIconPress: mockOnIconPress,
};

test('should match snapshot', () => {
  const tree = TestRenderer.create(<SocialLoginButtons {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('should render SocialLoginButtons', () => {
  render(<SocialLoginButtons {...props} />);
  const SocialLoginButtonsElement = screen.getByTestId(
    'test-socialloginbuttons-id',
  );
  expect(SocialLoginButtonsElement).toBeOnTheScreen();
});

test('should render SocialLoginButtons - render google button', () => {
  render(<SocialLoginButtons {...props} />);
  const SocialLoginButtonsElement = screen.getByTestId(
    'test-socialloginbuttons-id',
  );
  expect(SocialLoginButtonsElement).toBeOnTheScreen();
  const buttonElement = within(SocialLoginButtonsElement).getByTestId(
    'socialloginbuttons-button-google',
  );
  expect(buttonElement).toBeOnTheScreen();
  const buttonImgElement1 = within(buttonElement).getByTestId('img-google');
  expect(buttonImgElement1).toBeOnTheScreen();
});

test('should render SocialLoginButtons - render facebook button', () => {
  render(<SocialLoginButtons {...props} />);
  const SocialLoginButtonsElement = screen.getByTestId(
    'test-socialloginbuttons-id',
  );
  expect(SocialLoginButtonsElement).toBeOnTheScreen();
  const buttonElement = within(SocialLoginButtonsElement).getByTestId(
    'socialloginbuttons-button-facebook',
  );
  expect(buttonElement).toBeOnTheScreen();
  const buttonImgElement1 = within(buttonElement).getByTestId('img-facebook');
  expect(buttonImgElement1).toBeOnTheScreen();
});

const googleMockOnIconPress = jest.fn<
  SocialLoginButtonsProps['onIconPress'],
  [origin: string]
>();
test('should handle correct onPress on google button', () => {
  render(<SocialLoginButtons {...props} onIconPress={googleMockOnIconPress} />);
  const SocialLoginButtonsElement = screen.getByTestId(
    'test-socialloginbuttons-id',
  );
  expect(SocialLoginButtonsElement).toBeOnTheScreen();
  const buttonElement = within(SocialLoginButtonsElement).getByTestId(
    'socialloginbuttons-button-google',
  );
  expect(buttonElement).toBeOnTheScreen();
  const buttonImgElement1 = within(buttonElement).getByTestId('img-google');
  expect(buttonImgElement1).toBeOnTheScreen();
  expect(googleMockOnIconPress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(googleMockOnIconPress).toBeCalledTimes(1);
  expect(googleMockOnIconPress).toBeCalledWith('google');
});

const facebookMockOnIconPress = jest.fn<
  SocialLoginButtonsProps['onIconPress'],
  [origin: string]
>();
test('should handle correct onPress on facebook button', () => {
  render(
    <SocialLoginButtons {...props} onIconPress={facebookMockOnIconPress} />,
  );
  const SocialLoginButtonsElement = screen.getByTestId(
    'test-socialloginbuttons-id',
  );
  expect(SocialLoginButtonsElement).toBeOnTheScreen();
  const buttonElement = within(SocialLoginButtonsElement).getByTestId(
    'socialloginbuttons-button-facebook',
  );
  expect(buttonElement).toBeOnTheScreen();
  const buttonImgElement1 = within(buttonElement).getByTestId('img-facebook');
  expect(buttonImgElement1).toBeOnTheScreen();
  expect(facebookMockOnIconPress).toBeCalledTimes(0);
  fireEvent.press(buttonElement);
  expect(facebookMockOnIconPress).toBeCalledTimes(1);
  expect(facebookMockOnIconPress).toBeCalledWith('facebook');
});
