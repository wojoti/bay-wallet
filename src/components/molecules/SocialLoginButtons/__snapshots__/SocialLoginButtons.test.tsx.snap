// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "flex": undefined,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="test-socialloginbuttons-id"
>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "47%",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "borderColor": "#E2E2E2",
            "borderRadius": 6,
            "borderWidth": 1,
            "height": 60,
          },
        ]
      }
      testID="socialloginbuttons-button-google"
    >
      <SvgMock
        height={25}
        testID="img-google"
        width="100%"
      />
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "stretch",
          "flex": undefined,
          "flexDirection": "column",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        {
          "width": "47%",
        },
      ]
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        [
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "borderStyle": "solid",
            "justifyContent": "center",
            "opacity": 1,
          },
          false,
          {
            "borderColor": "#E2E2E2",
            "borderRadius": 6,
            "borderWidth": 1,
            "height": 60,
          },
        ]
      }
      testID="socialloginbuttons-button-facebook"
    >
      <SvgMock
        height={25}
        testID="img-facebook"
        width="100%"
      />
    </View>
  </View>
</View>
`;
